test/gtest/ucp/test_ucp_rma.cc

```diff
@@ -19,290 +23,143 @@ class test_ucp_rma : public test_ucp_memheap {
+                            ucs_memory_type_t target_mem_type,
+                            unsigned mem_map_flags) {
+        static const size_t MAX_SIZE = (100 * UCS_MBYTE) /
+                                       ucs::test_time_multiplier();
+        ucs::detail::message_stream ms("INFO");
+
+        ms << ucs_memory_type_names[send_mem_type] << "->" <<
+              ucs_memory_type_names[target_mem_type] << " ";
+        if (mem_map_flags & UCP_MEM_MAP_NONBLOCK) {
+            ms << "map_nb ";
+        }
 
-UCS_TEST_P(test_ucp_rma, nonblocking_stream_put_nb_flush_ep) {
-    test_nonblocking_implicit_stream_xfer(static_cast<nonblocking_send_func_t>(&test_ucp_rma::nonblocking_put_nb),
-                       DEFAULT_SIZE, DEFAULT_ITERS,
-                       1, false, true);
-    test_nonblocking_implicit_stream_xfer(static_cast<nonblocking_send_func_t>(&test_ucp_rma::nonblocking_put_nb),
-                       DEFAULT_SIZE, DEFAULT_ITERS,
-                       1, true, true);
-}
+        /* Test different random sizes */
+        for (size_t current_max_size = 128; current_max_size < MAX_SIZE;
+             current_max_size *= 4) {
 
-UCS_TEST_P(test_ucp_rma, nonblocking_get_nbi_flush_worker) {
-    test_blocking_xfer(static_cast<nonblocking_send_func_t>(&test_ucp_rma::nonblocking_get_nbi),
-                       DEFAULT_SIZE, DEFAULT_ITERS,
-                       1, false, false);
-    test_blocking_xfer(static_cast<nonblocking_send_func_t>(&test_ucp_rma::nonblocking_get_nbi),
-                       DEFAULT_SIZE, DEFAULT_ITERS,
-                       1, true, false);
-}
+            size_t size        = ucs::rand() % current_max_size;
+            unsigned num_iters = ucs_min(100, MAX_SIZE / (size + 1));
+            num_iters          = ucs_max(1, num_iters / ucs::test_time_multiplier());
 
-UCS_TEST_P(test_ucp_rma, nonblocking_get_nbi_flush_ep) {
-    test_blocking_xfer(static_cast<nonblocking_send_func_t>(&test_ucp_rma::nonblocking_get_nbi),
-                       DEFAULT_SIZE, DEFAULT_ITERS,
-                       1, false, true);
-    test_blocking_xfer(static_cast<nonblocking_send_func_t>(&test_ucp_rma::nonblocking_get_nbi),
-                       DEFAULT_SIZE, DEFAULT_ITERS,
-                       1, true, true);
-}
+            ms << num_iters << "x" << size << " ";
+            fflush(stdout);
 
-UCS_TEST_P(test_ucp_rma, nonblocking_stream_get_nbi_flush_worker) {
-    test_nonblocking_implicit_stream_xfer(static_cast<nonblocking_send_func_t>(&test_ucp_rma::nonblocking_get_nbi),
-                       DEFAULT_SIZE, DEFAULT_ITERS,
-                       1, false, false);
-    test_nonblocking_implicit_stream_xfer(static_cast<nonblocking_send_func_t>(&test_ucp_rma::nonblocking_get_nbi),
-                       DEFAULT_SIZE, DEFAULT_ITERS,
-                       1, true, false);
-}
+            ucs_memory_type_t mem_types[2] = {send_mem_type, target_mem_type};
```

Comment:
1. the size of the array can be removed
2. do we need whitespace after/before `{`/`}`?

Reply:
1. will remove
2. no, it's same as `(` and `[`